agent/engine/task_manager.go

```diff
@@ -449,6 +508,30 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 	}
 }
 
+// check if we need to restart the container
+func shouldRestartContainer(container *apicontainer.Container, containerChange dockerContainerChange) bool {
```

Comment:
i think this should be a method of container

Reply:
Will refactor this.

Reply:
Actually this file imports `container.go`, we'll have circular import error by changing this to a method of Container.

Reply:
why do you need to import task engine in container.go? i think you can just pass some strings as arguments and return a bool